cloud/deploy/deploy.go

```diff
@@ -801,3 +807,75 @@ func WarnIfNonLatestVersion(version string, httpClient *httputil.HTTPClient) {
 		fmt.Printf("WARNING! You are currently running Astro Runtime Version %s\nConsider upgrading to the latest version, Astro Runtime %s\n", version, latestRuntimeVersion)
 	}
 }
+
+// deployClientImage handles the client deploy functionality
+func deployClientImage(deployInput InputDeploy, c *config.Context) error { //nolint:gocritic
+	fmt.Println("Deploying client image...")
+
+	// Get the remote client registry endpoint from config
+	registryEndpoint := config.CFG.RemoteClientRegistry.GetString()
+	if registryEndpoint == "" {
+		return errors.New("remote client registry is not configured. Please run 'astro dev init' with remote execution enabled first, or configure it using 'astro config set remote.client_registry <endpoint>'")
+	}
+
+	// Determine the image tag to use
+	var imageTag string
+	if deployInput.ImageName != "" {
+		// Extract tag from custom image name if provided
+		parts := strings.Split(deployInput.ImageName, ":")
+		if len(parts) == 2 {
+			imageTag = parts[1]
+		} else {
+			imageTag = "latest"
+		}
+	} else {
+		// Use default deploy-<timestamp> tagging mechanism
+		imageTag = "deploy-" + time.Now().UTC().Format("2006-01-02T15-04")
+	}
+
+	// Build the full remote image name
+	remoteImage := fmt.Sprintf("%s:%s", registryEndpoint, imageTag)
+
+	// Create an image handler for building and pushing
+	localImageName := airflow.ImageName("agent-client", "latest")
+	imageHandler := airflowImageHandler(localImageName)
+
+	if deployInput.ImageName != "" && strings.Contains(deployInput.ImageName, ":") {
+		// If a full image name with tag is provided, use it directly
+		fmt.Println("Using provided image:", deployInput.ImageName)
+		err := imageHandler.TagLocalImage(deployInput.ImageName)
+		if err != nil {
+			return fmt.Errorf("failed to tag local image: %w", err)
+		}
+	} else {
+		// Authenticate with the base image registry before building
+		// This is needed because Dockerfile.client uses base images from a private registry
+		baseImageRegistry := "images.astronomer.cloud"
+		err := airflow.DockerLogin(baseImageRegistry, registryUsername, c.Token)
+		if err != nil {
+			return fmt.Errorf("failed to authenticate with registry %s: %w", baseImageRegistry, err)
+		}
+
+		// Build the client image from the current directory
+		fmt.Println("Building client image...")
+		buildConfig := types.ImageBuildConfig{
+			Path:            deployInput.Path,
+			TargetPlatforms: deployImagePlatformSupport,
```

Comment:
I think we should be open to allowing image build for other platforms via the input flag only when building client images